extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayMultiDatasourceRuntimeConfig.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.flyway.runtime;
+
+import java.util.Collections;
+import java.util.Map;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "flyway", phase = ConfigPhase.RUN_TIME)
+public final class FlywayMultiDatasourceRuntimeConfig {
+    /*
+     * Creates a {@link FlywayMultiDatasourceRuntimeConfig} with default settings.
+     * 
+     * @return {@link FlywayMultiDatasourceRuntimeConfig}
+     */
+    public static final FlywayMultiDatasourceRuntimeConfig defaultConfig() {
+        return new FlywayMultiDatasourceRuntimeConfig();
+    }
+
+    /**
+     * Gets the {@link FlywayRuntimeConfig} for the given datasource name.<br>
+     * The name of the default datasource is an empty {@link String}.
+     * 
+     * @param dataSourceName {@link String}
+     * @return {@link FlywayRuntimeConfig}
+     * @throws NullPointerException if dataSourceName is null.
+     */
+    public FlywayRuntimeConfig getConfigForDataSourceName(String dataSourceName) {
+        if (dataSourceName.isEmpty()) {
+            return defaultDataSource;
+        }
+        return namedDataSources.getOrDefault(dataSourceName, FlywayRuntimeConfig.defaultConfig());
```

Comment:
Hmmm, so now I understand this default thing, I will remove the other comments on the subject. I'm not very excited about it but I don't see a better way to do it given all the Flyway config is optional.

Reply:
I left out the comments regarding the default datasource. If you want, i could do this in a separate PR.